### PR TITLE
Fix sample barcodes

### PR DESCRIPTION
**Pull request by alexvpickering**
# Background
#### Link to issue 

image

https://biomage.atlassian.net/secure/RapidBoard.jspa?rapidView=1&projectKey=BIOMAGE&modal=detail&selectedIssue=BIOMAGE-734

#### Anything else the reviewers should know about the changes here

These are fixes related to a unisample dataset comming from data-ingest and includes:

1) FDR for the sample is in a sample-specific subitem of the config
2) The sample-id is no longer prepended to the barcodes but in the `metadata$samples` (or at least not for unisample datasets). However we still need to support this until old data objects are re-ingested.
3) Seurat was throwing an error when it was not attached (see above image)

# Changes
### Code changes

# Definition of DONE
Your changes will be ready for merging after each of the steps below have been completed:

### Testing
- [ ] Unit tests written
- [ ] Tested locally with Inframock
- [ ] Deployed to staging

To set up easy local testing with inframock, follow the instructions here: https://github.com/biomage-ltd/inframock
To deploy to the staging environment, follow the instructions here: https://github.com/biomage-ltd/biomage-utils

### Documentation updates
Is all relevant documentation updated to reflect the proposed changes in this PR?

- [ ] Relevant Github READMEs updated
- [ ] Relevant wiki pages created/updated

### Approvers
- [ ] Approved by a member of the core engineering team
- [ ] (UX changes) Approved by vickymorrison (this is her username, tag her if you need approval)

### Just before merging:
- [ ] After the PR is approved, the `unstage` script in here: https://github.com/biomage-ltd/biomage-utils is executed. This script cleans up your deployment to staging

### Optional
- [ ] Photo of a cute animal attached to this PR

### CODE REVIEW

**Review by marcellp — COMMENTED** [review]
Thanks for this fix re the null string, looks good to me. Can a bioinformatician also review the rest of the code for correctness regarding the data format?

**Review by marcellp — APPROVED** [review]

File: qc-runner/src/classifier.r [hunk 1]
```diff
@@ -69,7 +70,7 @@ task <- function(seurat_obj, config, task_name, sample_id){
 
   # Check wheter the filter is set to true or false
   # For some reason the last children of named lists are computed as vectors, so we can't access them as recursive objects. 
-  FDR <- as.numeric(config$filterSettings[["FDR"]])
+  FDR <- as.numeric(config$filterSettings[[tmp_sample]][["FDR"]])
```
Comment (xverges) [inline qc-runner/src/classifier.r:73]: Is this always defined? Do we always send per-sample configuration?

Comment (ogibson-biomage) [reply qc-runner/src/classifier.r:73]: Actually the configuration list should be sample-specific to the current sample. Besides, tmp_sample is not defined at this point in the execution, we should look into this.

**Review by ogibson-biomage — CHANGES_REQUESTED** [review]

File: qc-runner/src/classifier.r (hunk 1 above)
Comment (xverges) [reply qc-runner/src/classifier.r:73]: `tmp_sample <- sub("sample-","",sample_id)` :-)

Comment (ogibson-biomage) [reply qc-runner/src/classifier.r:73]: Hmm I can't find the line

Comment (ogibson-biomage) [reply qc-runner/src/classifier.r:73]: Oh I hadn't collapsed the line

**Review by JuanluOnieva-biomage — APPROVED** [review]

**Review by ogibson-biomage — APPROVED** [review]
Looks good

**Pull request merged**